cdptools/pipelines/event_gather_pipeline.py

```diff
@@ -124,45 +146,62 @@ def task_transcript_get_or_create(self, key: str, audio_uri: str, phrases: Optio
             file_store=self.file_store,
             algorithm_name="EventGatherPipeline.task_transcript_get_or_create",
             algorithm_version=get_module_version(),
-            inputs=[key, audio_uri],
+            inputs=[key, video_uri, caption_uri],
             remove_files=True
         ) as run:
-            # Setup temporary filenames
-            tmp_raw_transcript_filepath = f"{key}_raw_transcript_0.json"
-            tmp_ts_words_transcript_filepath = f"{key}_ts_words_transcript_0.json"
-            tmp_ts_sentences_transcript_filepath = f"{key}_ts_sentences_transcript_0.json"
-
-            # Check if raw transcript already exists in file store
-            try:
-                # Try get the best variant of transcript
-                # Timestamped sentences first
+            # Setup temporary filenames, sorted by preference
+            tmp_filenames = [
+                f"{key}_ts_speaker_turns_transcript_0.json",
+                f"{key}_ts_sentences_transcript_0.json",
+                f"{key}_ts_words_transcript_0.json",
+                f"{key}_raw_transcript_0.json"
+            ]
+
+            # Iterate through tmp_filenames and check if any of them exists
+            main_transcript_uri = None
```

Comment:
This is a better implementation. Nice job.

Reply:
Yea, a nested block of 4 try/exceptions is a bit too hard to read/understand.